JASP-Engine/JASP/R/bainanovabayesian.R

```diff
@@ -125,181 +119,193 @@ BainAnovaBayesian <- function (jaspResults, dataset, options, ...) {
-					d <- data.frame(x=-Inf, xend=-Inf, y=min(b), yend=max(b))
-					list(ggplot2::geom_segment(data=d, ggplot2::aes(x=x, y=y, xend=xend, yend=yend), inherit.aes=FALSE, size = 1),
-						 ggplot2::scale_y_continuous(breaks=c(min(b),max(b))))
-					 }
-			groupVars <- options$fixedFactors
-			groupVars <- unlist(groupVars)
-
-			groupVarsV <- .v(groupVars)
-			dependentV <- .v(options$dependent)
-
-			sum_model <- bainResult$estimate_res
-			summaryStat <- summary(sum_model)$coefficients
-
-			if(type == "ancova"){
-				summaryStat <- summaryStat[-(nrow(summaryStat) - 0:(length(options[["covariates"]])-1)), ] # Remove covars rows
-			}
-
-			summaryStat <- cbind(summaryStat, 1:nrow(summaryStat))
-			colnames(summaryStat)[length(colnames(summaryStat))] <- "plotHorizontalAxis"
-			colnames(summaryStat)[which(colnames(summaryStat) == "Estimate")] <- "dependent"
-			summaryStatSubset <- as.data.frame(summaryStat)
-
-			groupCol <- dataset[ , .v(options[["fixedFactors"]])]
-			varLevels <- levels(groupCol)
-			ciLower <- summaryStatSubset[, 1] - 1.96*summaryStatSubset[, 2]
-			ciUpper <- summaryStatSubset[, 1] + 1.96*summaryStatSubset[, 2]
-			summaryStatSubset$ciLower <- ciLower
-			summaryStatSubset$ciUpper <- ciUpper
-			summaryStat <- summaryStatSubset
-
-				p <- ggplot2::ggplot(summaryStatSubset, ggplot2::aes(x=plotHorizontalAxis,
-											y=dependent,
-											group=1))
-
-
-				pd <- ggplot2::position_dodge(.2)
-				p = p + ggplot2::geom_errorbar(ggplot2::aes(ymin=ciLower,
-															ymax=ciUpper),
-															colour="black", width=.2, position=pd)
-
-
-			p <- p + ggplot2::geom_line(position=pd, size = .7) +
-				ggplot2::geom_point(position=pd, size=4) +
-				ggplot2::scale_fill_manual(values = c(rep(c("white","black"),5),rep("grey",100)), guide=ggplot2::guide_legend(nrow=10)) +
-				ggplot2::scale_shape_manual(values = c(rep(c(21:25),each=2),21:25,7:14,33:112), guide=ggplot2::guide_legend(nrow=10)) +
-				ggplot2::scale_color_manual(values = rep("black",200),guide=ggplot2::guide_legend(nrow=10)) +
-				ggplot2::ylab(options$dependent) +
-				ggplot2::xlab(groupVars) +
-				base_breaks_y(summaryStat, TRUE)
-			p <- JASPgraphs::themeJasp(p)
-
-				jaspResults[["descriptivesPlot"]] <- createJaspPlot(plot = p, title = "Descriptives Plot")
-				jaspResults[["descriptivesPlot"]]$dependOn(options =c("descriptivesPlot", "fixedFactors", "dependent", "model"))
-				jaspResults[["descriptivesPlot"]]$position <- 4
-		}
-	} else if(options[["descriptivesPlot"]]){
-			errorPlot <- createJaspPlot(plot = NULL, title = "Descriptives Plot")
-			errorPlot$setError("Plotting not possible: No analysis has been run.")
-			jaspResults[["descriptivesPlot"]] <- errorPlot
-			jaspResults[["descriptivesPlot"]]$dependOn(options =c("descriptivesPlot", "fixedFactors", "dependent", "model"))
-			jaspResults[["descriptivesPlot"]]$position <- 4
+.bainAnovaDescriptivesPlot <- function(dataset, options, bainContainer, ready, type = "anova") {
```

Comment:
Does this need to be very different from the descriptives plot we already have in the other analyses?

Reply:
No idea